bool[8] channel
